rockcraft/extensions/gunicorn.py

```diff
@@ -117,6 +117,12 @@ def get_root_snippet(self) -> Dict[str, Any]:
           - services: a service to run the Gunicorn server
           - parts: see _GunicornBase._gen_parts
         """
+        for required_key in ("base", "build-base", "platforms"):
```

Comment:
The rules for `build-base` are [a bit more complicated](https://github.com/canonical/rockcraft/blob/main/rockcraft/models/project.py#L170) than this: if `build-base` is missing, it's the value of `base`, *unless* `base` is `"bare"` (in which case this is an error).
I'm thinking now that we probably want to drop this validation here altogether, because these fields [are already validated](https://github.com/canonical/rockcraft/blob/main/rockcraft/models/project.py#L136) before getting here. The new flow in the application is this:
- A `BuildPlanner` is built from the "unprocessed" yaml data to determine the base and what to build (the platforms). This validates the base, build-base and platforms entries;
- The original yaml is processed by applying extensions, global environment, grammar, etc. This step is where this method would be called;
- The processed yaml is then used to create a regular `Project` model.

Another argument for not adding the validation here is that we'd have to keep the two validations "in sync" - I think it's better to enforce "by contract" that the yaml will always have the things the project needs.

Reply:
so it sounds like we just need to remove the logic that adds the fields

Reply:
ok that should be fixed now, I looked through the docs and couldn't find any reference to our treatment of `base`, `build-base` and `platforms`